File: Mod09/Labfiles/Starter/CS/File_IO_CS/File_IO_CS/Person.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace File_IO_CS
{
    abstract class Person
    {
        private float height;
        private float weight;
        private string gender;
        private int age;
        private string firstName;
        private string lastName;

        public float Height
        {
            get { return height; }
            set { height = value; }
        }

        public float Weight
        {
            get { return weight; }
            set { weight = value; }
        }

        public string Gender
        {
            get { return gender; }
            set { gender = value; }
        }

        public int Age
        {
            get { return age; }
            set { age = value; }
        }

        public string FirstName
        {
            get { return firstName; }
            set { firstName = value; }
        }

        public string LastName
        {
            get { return lastName; }
            set { lastName = value; }
        }


        // public methods
        public virtual void eat()
        {
            Console.WriteLine("slurping");
        }

        public void sleep()
        {
            Console.WriteLine("Snoring");
        }

        abstract public void move();

        public void communicate()
        {

        }
    }
}

File: Mod07/Democode/VB/PolymorphismDemo_VB/PolymorphismDemo_VB/Person.vb
Public Class Person

    Private _height As Single
    Private _weight As Single
    Private _gender As String
    Private _age As Integer
    Private _firstName As String
    Private _lastName As String

    Public Property Height As Single
        Get
            Return Me._height
        End Get
        Set(value As Single)
            Me._height = value
        End Set
    End Property

    Public Property Weight As Single
        Get
            Return Me._weight
        End Get
        Set(value As Single)
            Me._weight = value
        End Set
    End Property

    Public Property Age As Integer
        Get
            Return Me._age
        End Get
        Set(value As Integer)
            Me._age = value
        End Set
    End Property

    Public Property Gender As String
        Get
            Return Me._gender
        End Get
        Set(value As String)
            Me._gender = value
        End Set
    End Property

    Public Property FirstName As String
        Get
            Return Me._firstName
        End Get
        Set(value As String)
            Me._firstName = value
        End Set
    End Property

    Public Property LastName As String
        Get
            Return Me._lastName
        End Get
        Set(value As String)
            Me._lastName = value
        End Set
    End Property

    Public Overridable Sub eat()

    End Sub

    Public Sub sleep()

    End Sub

    Public Overridable Sub move()

    End Sub

    Public Sub communicate()

    End Sub

End Class
